omod/src/main/webapp/fragments/patientdashboard/visitIncludes.gsp

```diff
@@ -73,19 +73,33 @@
             ])}
         </p>
 
-        <p>
-            <label for="stopDate" class="required">
+        <p>          
+                <label for="stopDate" class="required">
                 ${ ui.message("coreapps.stopDate.label") }
-            </label>
+                </label>
 
-            ${ ui.includeFragment("uicommons", "field/datetimepicker", [
+ <% if(activeVisits){ %>
+                ${ ui.includeFragment("uicommons", "field/datetimepicker", [
                     id: "retrospectiveVisitStopDate",
                     formFieldName: "retrospectiveVisitStopDate",
                     label:"",
                     defaultDate: visitEndTime,
                     endDate: editDateFormat.format(visitEndTime),
                     useTime: false,
-            ])}
+                ])}
+
+            <% } else { %>
+
+            ${ ui.includeFragment("uicommons", "field/datetimepicker", [
+                    id: "noActiveVisitStopDate",
```

Comment:
Why are we adding this field/fragment?

Reply:
From line 81. This is to force the user have a default stop date if they are entering a retrospective Visit yet we are currently having an active visit otherwise if there is no active visit then we can allow them create a retrospective visit that has a null end date thus making it Active(there is already a catch implemented if the restrospective visit is overlaping with another)